<!-- Launcher for arduino microcontroller with rosserial-->

<launch>
  
  <!-- Load the controller parameters -->
  <rosparam 
      file="$(find ghost)/param/arduino_controller_params.yaml"
      command="load" 
  />
  
  <!-- Node for converter CarControl messages to ServoControl messages -->
  <node
    pkg="ghost"
    type="car_command_converter"
    name="car_command_converter"
  />
  
  <!-- rosserial node 
     
     Argument sets serial port
        Android Uno:  /dev/ttyACM0
        Android Nano: /dev/ttyUSB0
  -->
  <node
    pkg="rosserial_python"
    type="serial_node.py"
    name="arduino_serial"
    args="/dev/ttyUSB0">
      <param name="baud" value="115200"/>
  </node>

  <!-- Load URDF file -->
  <include file="$(find ghost)/launch/includes/state_publisher.launch.xml" />
  
</launch>
